source/extensions/filters/http/transformation/inja_transformer.cc

```diff
@@ -80,7 +102,34 @@ Extractor::extract(Http::StreamFilterCallbacks &callbacks,
     if (header_entries.empty()) {
       return "";
     }
-    return extractValue(callbacks, header_entries[0]->value().getStringView());
+    const auto &header_value = header_entries[0]->value().getStringView();
+    return extractValue(callbacks, header_value);
+  }
+}
+
+std::string
+Extractor::extractDestructive(Http::StreamFilterCallbacks &callbacks,
+                   const Http::RequestOrResponseHeaderMap &header_map,
+                   GetBodyFunc &body) const {
+  if (body_) {
+    const std::string &string_body = body();
+    absl::string_view sv(string_body);
+    if (mode_ == ExtractionApi::SINGLE_REPLACE) {
+      return replaceIndividualValue(callbacks, sv);
+    } else {
+      return replaceAllValues(callbacks, sv);
+    }
+  } else {
+    const Http::HeaderMap::GetResult header_entries = getHeader(header_map, headername_);
+    if (header_entries.empty()) {
+      return "";
+    }
+    const auto &header_value = header_entries[0]->value().getStringView();
+    if (mode_ == ExtractionApi::SINGLE_REPLACE) {
+      return replaceIndividualValue(callbacks, header_value);
+    } else {
```

Comment:
It might be good to be explicit about our `mode_` expectation in the else case to guard against future API changes missing updating this code. In fact this should probably be a switch statement at that point

Reply:
It definitely should. I ran into this issue while writing unit tests and let it slide, so thanks for the catch!